Library/Homebrew/extend/cachable.rb

```diff
@@ -11,4 +11,43 @@ def cache
   def clear_cache
     cache.clear
   end
+
+  # Collect all classes that mix in Cachable so that those caches can be cleared in-between tests.
+  if ENV["HOMEBREW_TESTS"]
```

Comment:
I wonder, could we implement this as a test helper with `allow(Cachable).to receive(:extended)`.

Reply:
I don't think so. It needs to be set up before other classes start mixing in this module. I thought of another way of doing this where we load a separate file in `test/support/extend/` with these cachable methods before loading `global.rb` and then we can be sure that this only gets loaded in the test environment.

Reply:
See https://github.com/Homebrew/brew/pull/16746/commits/909828caa1565da181aa877613c5992efa860c99